surefire-providers/surefire-junit-platform/src/main/java/org/apache/maven/surefire/junitplatform/JUnitPlatformProvider.java

```diff
@@ -275,11 +275,11 @@ private LauncherDiscoveryRequest buildLauncherDiscoveryRequestForRerunFailures(
                 .map( TagFilter::excludeTags )
                 .ifPresent( filters::add );
 
-        TestListResolver testListResolver = parameters.getTestRequest().getTestListResolver();
-        if ( !testListResolver.isEmpty() )
-        {
-            filters.add( new TestMethodFilter( testListResolver ) );
-        }
+        of( optionallyWildcardFilter( parameters.getTestRequest().getTestListResolver() ) )
+            .filter( f -> !f.isEmpty() )
+            .filter( f -> !f.isWildcard() )
+            .map( TestMethodFilter::new )
+            .ifPresent( filters::add );
 
```

Comment:
So when you agree with my next commit can you use `hasMethodPatterns()` instead of `optionallyWildcardFilter`
it will be removed, and I will need fix this line also.

Reply:
That's trivial change while rebasing. I also have to do it sometime.